[GStreamer] SVT-AV1 encoder support

philn opened the pull request:
#### 9bd96d9df3ee495ca17cb8d930502e52144af6f7
<pre>
[GStreamer] SVT-AV1 encoder support
<a href="https://bugs.webkit.org/show_bug.cgi?id=256960">https://bugs.webkit.org/show_bug.cgi?id=256960</a>

Reviewed by Xabier Rodriguez-Calvar.

Also enable QoS so that the encoder can adapt in case of Quality of Service events coming from
downstream elements.

* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(videoEncoderSetEncoder):
(webkit_video_encoder_class_init):

Canonical link: <a href="https://commits.webkit.org/297125@main">https://commits.webkit.org/297125@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80aec99f70a84c281db877845fdc2eb42f323ca1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110579 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30238 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20671 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116605 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60846 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112542 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30917 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38826 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84089 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113527 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24689 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99572 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64530 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24052 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17712 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60400 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94074 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17771 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119395 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37619 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27936 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93055 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37993 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95843 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92877 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23664 "Found 1 new test failure: http/tests/workers/service/registerServiceWorkerClient-after-network-process-crash.html (failure)") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37889 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15632 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33593 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17849 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37514 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42985 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37177 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40516 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38885 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/484a69f134d974ee117465ee7cc90cd923c5c9ba. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47650

calvaris commented on Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
```diff
@@ -359,6 +360,8 @@ static bool videoEncoderSetEncoder(WebKitVideoEncoder* self, EncoderId encoderId
     }
 
     encoderDefinition->setupEncoder(self);
+    g_object_set(priv->encoder.get(), "qos", TRUE, nullptr);
```
Do all encoders have the qos property. Wouldn't be a good idea to check for it?

philn replied:
Well it is in the base encoder class and none of the encoders we wrap currently inherits from a different class.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/70fadaf5b1f6ace7ab4f88b47dba42274a62b086. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47650

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/80aec99f70a84c281db877845fdc2eb42f323ca1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47650

philn committed:
[GStreamer] SVT-AV1 encoder support
https://bugs.webkit.org/show_bug.cgi?id=256960

Reviewed by Xabier Rodriguez-Calvar.

Also enable QoS so that the encoder can adapt in case of Quality of Service events coming from
downstream elements.

* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(videoEncoderSetEncoder):
(webkit_video_encoder_class_init):

Canonical link: https://commits.webkit.org/297125@main

webkit-commit-queue commented:
Committed 297125@main (9bd96d9df3ee): <https://commits.webkit.org/297125@main>

Reviewed commits have been landed. Closing PR #47650 and removing active labels.
